Objects/dictobject.c

```diff
@@ -1951,6 +1951,17 @@ build_indices_unicode(PyDictKeysObject *keys, PyDictUnicodeEntry *ep, Py_ssize_t
     }
 }
 
+
+static void
+invalidate_and_clear_inline_values(PyDictValues *values)
+{
+    assert(values->embedded);
+    FT_ATOMIC_STORE_UINT8(values->valid, 0);
+    for (int i = 0; i < values->capacity; i++) {
+        FT_ATOMIC_STORE_PTR(values->values[i], NULL);
```

Comment:
This can be `FT_ATOMIC_STORE_PTR_RELEASE`.

I think the write to `valid` can be weakened too, but I'm less sure about the ordering and it's only one write (instead of up to 30) so I'm not going to worry about it.